tests/ArmTemplates.Tests/Extractor/Scenarios/ApiExtractorTests.cs

```diff
@@ -138,5 +139,84 @@ public async Task GenerateApiTemplates_ProperlyLaysTheInformation()
+            var mockedTagClient = MockTagClient.GetMockedApiClientWithDefaultValues();
+            var mockedApiOperationClient = MockApiOperationClient.GetMockedApiClientWithDefaultValues();
+            var mockedDiagnosticClient = MockDiagnosticClient.GetMockedClientWithApiDependentValues();
+
+            // mocked extractors
+            var mockedDiagnosticExtractor = new DiagnosticExtractor(this.GetTestLogger<DiagnosticExtractor>(), mockedDiagnosticClient);
+            var mockedApiSchemaExtractor = new ApiSchemaExtractor(this.GetTestLogger<ApiSchemaExtractor>(), mockedApiSchemaClient);
+            var mockedPolicyExtractor = new PolicyExtractor(this.GetTestLogger<PolicyExtractor>(), mockedPolicyClient, new TemplateBuilder());
+            var mockedProductApisExtractor = new ProductApisExtractor(this.GetTestLogger<ProductApisExtractor>(), mockedProductClient, mockedApiClient, new TemplateBuilder());
+            var mockedTagExtractor = new TagExtractor(this.GetTestLogger<TagExtractor>(), mockedTagClient, new TemplateBuilder());
+            var mockedApiOperationExtractor = new ApiOperationExtractor(this.GetTestLogger<ApiOperationExtractor>(), mockedApiOperationClient);
+
+            var apiExtractor = new ApiExtractor(
+                this.GetTestLogger<ApiExtractor>(),
+                new TemplateBuilder(),
+                mockedApiClient,
+                mockedDiagnosticExtractor,
+                mockedApiSchemaExtractor,
+                mockedPolicyExtractor,
+                mockedProductApisExtractor,
+                mockedTagExtractor,
+                mockedApiOperationExtractor);
+
+            var extractorExecutor = ExtractorExecutor.BuildExtractorExecutor(
+                this.GetTestLogger<ExtractorExecutor>(),
+                apiExtractor: apiExtractor);
+            extractorExecutor.SetExtractorParameters(extractorParameters);
+
+            // act
+            var apiTemplate = await extractorExecutor.GenerateApiTemplateAsync(
+                singleApiName: It.IsAny<string>(),
+                multipleApiNames: It.IsAny<List<string>>(),
+                currentTestDirectory);
+
+            // assert
+            File.Exists(Path.Combine(currentTestDirectory, apiTemplate.TypedResources.FileName)).Should().BeTrue();
+            Directory.GetFiles(Path.Combine(currentTestDirectory, PolicyExtractor.PoliciesDirectoryName)).Count().Should().Be(4);
+
+            apiTemplate.Parameters.Should().NotBeNull();
+            apiTemplate.Parameters.Should().ContainKey(ParameterNames.ApimServiceName);
+            apiTemplate.Parameters.Should().ContainKey(ParameterNames.ServiceUrl);
+            apiTemplate.Parameters.Should().ContainKey(ParameterNames.ApiLoggerId);
+            apiTemplate.Parameters.Should().ContainKey(ParameterNames.PolicyXMLBaseUrl);
+            apiTemplate.Parameters.Should().ContainKey(ParameterNames.PolicyXMLSasToken);
+            apiTemplate.Resources.Count().Should().Be(23);
```

Comment:
These checks are already existed in GenerateApiTemplates_ProperlyLaysTheInformation, we could test only specific newly introduced feature, in this case only ApiSchema client was changed, which does not change behaviour of other functionality

Reply:
removed un necessary validations